unit-tests/database_test.py

```diff
@@ -79,7 +79,15 @@ def test_iteration(self):
 
 class TestSQLiteStateDatabase(object):
     def test_constructor(self, tmp_path: Path):
-        _ = SqliteStateDatabase(tmp_path)
+        database = SqliteStateDatabase(tmp_path)
+
+        assert database._connection is None
+        assert database._working_directory == tmp_path
```

Comment:
Very uneasy with dipping into the classes "private" variables for testing. Test units should be treated as a black box.

Reply:
Does that suggest that we ought to drop testing of the constructor completely following this change then?  As post change I don't think there are any testable effects of the constructor?

Reply:
Testing can be hard. I have checked the coverage and your new tests at last visit every part of the SQLite code. As such I am happy for them to go ahead. We may want to revisit this issue later but that is for a new ticket.